ethereum/client.go

```diff
@@ -191,3 +371,32 @@ func (c *Client) WaitForConfirmations(tx *types.Transaction) error {
 	fmt.Println("tx confirmed")
 	return nil
 }
+
+func getRentCost(en *ens.Name, rentSeconds int64) (*big.Int, error) {
+	costSec, err := en.RentCost()
+	if err != nil {
+		return nil, err
+	}
+	return new(big.Int).Mul(costSec, big.NewInt(rentSeconds)), nil
+}
+
+// GetCombinedName is used to return a combined ens name
+func (c *Client) GetCombinedName(subName, parentName string) string {
+	// ensure that if the first character is not a .
+	// we make it a .
+	if parentName[0] != '.' {
+		parentName = "." + parentName
+	}
+	// if the subName includes a period, remove it
+	if subName[len(subName)-1] == '.' {
+		subName = strings.TrimSuffix(subName, ".")
+	}
+	// return the combined subname and parent name
+	return subName + parentName
+}
+
+func (c *Client) getGasPrice() *big.Int {
+	gprice := new(big.Int)
+	gprice.SetString("5000000000", 10)
```

Comment:
How is this decided?

Reply:
5gwei is generally a pretty safe average gas price to confirm transactions in a reasonable time 

https://www.ethgasstation.info/ is a nice website to estimate this kind of stuff, however eventually we will need to have a much more dynamic and flexible gas price calculation

Reply:
5gwei (this value) is generally a pretty safe average gas price to have transactions processed in a reasonable amount of time. Eventually we will want to try and use a more dynamic gas price.

The main issue with dynamic pricing is it will increase the cost of transactions. Because ENS usage is not a service we are charging for, and is instead something we only require users to pay gas costs via credits, a static 5gwei allows us to get reasonable costs.